### ISSUE TITLE
I can't find data prepared part recipe codes about sond diarization model

### ISSUE THREAD

**Issue by shanguanma** [modelscope/FunASR#1916]
Notice: In order to resolve issues more efficiently, please raise issue following the template.
（注意：为了更加高效率解决您遇到的问题，请按照模板提问，补充细节）

## ❓ Questions and Help
@[ZhihaoDU](https://github.com/modelscope/FunASR/commits?author=ZhihaoDU)
As the title suggests, I am reading your SOND model recently and I am very interested in it. Then I found that the link to the open source code provided in your paper is invalid. I cannot understand the details of the model, especially the details of the data processing part. Can you open source the relevant recipe code details? Thank you．

For example, section 2.1 in your paper,
`In our experiments, VAD results have been already provided by the corpus for fair evaluation, therefore, we directly use the
official VAD results in our system.`  Do I understand that you did not perform any additional VAD operations on the training set, development set, and test set of alimeeting (such as using the pre-trained VAD model provided by Kaldi.i.e.:http://kaldi-asr.org/models/12/0012_sad_v1.tar.gz), but only used the textgrid information provided by alimeeting to remove audio silence.

### Before asking:
1. search the issues.
2. search the docs.

#### What's your environment?

 - OS (e.g., Linux):
 - FunASR Version (e.g., 1.0.0):
 - ModelScope Version (e.g., 1.11.0):
 - PyTorch Version (e.g., 2.0.0):
 - How you installed funasr (`pip`, source):
 - Python version:
 - GPU (e.g., V100M32)
 - CUDA/cuDNN version (e.g., cuda11.7):
 - Docker version (e.g., funasr-runtime-sdk-cpu-0.4.1)
 - Any other relevant information:

**Comment by shanguanma** [comment]
No comments or responses ？

**Comment by LauraGPT** [comment]
You could ref to branch-0.8.8

**Comment by shanguanma** [comment]
> You could ref to branch-0.8.8

Thansk for your reply, I will learn from it. If I have any questions, I will ask you here.

**Comment by shanguanma** [comment]
Yes, I have follows your suggestion, the details are as follows:
1) I refer https://github.com/modelscope/FunASR/blob/v0.8.8/egs/alimeeting/diarization/sond/run.sh, I can use the pretrained sond model and preprocessed alimeeting test set to get the DER claimed in the paper(`https://arxiv.org/pdf/2211.10243`) of 4.12% on alimeeting test set. however, How to get preprocessed test data? The relevant code is missing in the current folder.
2) I refer https://github.com/modelscope/FunASR/blob/v0.8.8/egs/alimeeting/modular_sa_asr/run_diar.sh, For obtaining speaker profile, It uses Vbx to get the first diarization, then performs rttm2segment and remove overlap operations, obtains xvector on this basis, and finally performs resegment_data operation. The segments and wav.scp contained in the data_source_dir in this step are not clearly stated. I made two assumptions here. Assumption 1, it comes directly from the segments and wav.scp after using pretrain vad, and finally obtains DER 10.69% on alimeeting eval. Assumption 2, it comes directly from oracle segments (I use https://github.com/modelscope/FunASR/blob/v0.8.8/egs/alimeeting/sa_asr/run.sh --stage1 ---stop-stage 1 to obtain data/org/Eval_Ali_far/{segments,wav.scp} as data_source_dir, and finally obtain DER 10.14% on alimeeting eval) , However, It still does the same operation mentioned in the paper to get the speaker profile by using blstm spectral cluster, and the DER here is much different from what is claimed in the paper.

**Comment by shanguanma** [comment]
@LauraGPT ,Please reopen this issue.

**Comment by ZhihaoDU** [comment]
Yes, you are right. Since Alimeeting offers the oracle VAD information in textgrid format, we directly use it and exclude the impact of inaccurate VAD results in diarization tasks.

**Comment by shanguanma** [comment]
> Yes, you are right. Since Alimeeting offers the oracle VAD information in textgrid format, we directly use it and exclude the impact of inaccurate VAD results in diarization tasks.

Thanks for your reply, I followed your suggestion (https://github.com/modelscope/FunASR/issues/1959#issuecomment-2255721969), However, I can't reproduce your paper result. I hope that you could help me to solve my issue. I raised a new issue (https://github.com/modelscope/FunASR/issues/1959), I posted the script that reproduces your results in this issue. Please take the time to check it and point out any mistakes I made. Thank you again.
